### PR TITLE
Added the passing of user data for syscall events.

### PR DESCRIPTION
**Pull request by johnfxgalea**
Adds the passing of user data for callbacks pertaining to syscall events. 

See #2833

### CODE REVIEW

File: api/docs/release.dox [hunk 1]
```diff
@@ -219,6 +219,10 @@ Further non-compatibility-affecting changes include:
    physical cpus.
  - Added #DR_DISALLOW_UNSAFE_STATIC and dr_disallow_unsafe_static_behavior()
    for sanity checks to help support statically-linked clients.
+ - Added drmgr_register_pre_syscall_event_user_data() and
+   drmgr_unregister_pre_syscall_event_user_data to enable passing of user data.
```
Comment (derekbruening) [inline api/docs/release.dox]: Include () to get doxygen link

File: api/docs/release.dox [hunk 2]
```diff
@@ -219,6 +219,10 @@ Further non-compatibility-affecting changes include:
...
+ - Added drmgr_register_post_syscall_event_user_data() and
+   drmgr_unregister_post_syscall_event_user_data to enable passing of user data.
```
Comment (derekbruening) [inline api/docs/release.dox]: Include () to get doxygen link

File: api/docs/release.dox (hunk 1 above)
Comment (johnfxgalea) [reply api/docs/release.dox]: Done

File: api/docs/release.dox (hunk 2 above)
Comment (johnfxgalea) [reply api/docs/release.dox]: Done

**Comment by derekbruening** [comment]
run aarch64 tests

**Pull request merged**